modules/cudaoptflow/samples/optical_flow.cpp

```diff
@@ -177,12 +177,14 @@ int main(int argc, const char* argv[])
 
     GpuMat d_flow(frame0.size(), CV_32FC2), d_flowxy;
 
+    Stream inputStream, outputStream;
+
     Ptr<cuda::BroxOpticalFlow> brox = cuda::BroxOpticalFlow::create(0.197f, 50.0f, 0.8f, 10, 77, 10);
     Ptr<cuda::DensePyrLKOpticalFlow> lk = cuda::DensePyrLKOpticalFlow::create(Size(7, 7));
     Ptr<cuda::FarnebackOpticalFlow> farn = cuda::FarnebackOpticalFlow::create();
     Ptr<cuda::OpticalFlowDual_TVL1> tvl1 = cuda::OpticalFlowDual_TVL1::create();
     Ptr<cuda::NvidiaOpticalFlow_1_0> nvof = cuda::NvidiaOpticalFlow_1_0::create(frame0.size().width, frame0.size().height,
-        NvidiaOpticalFlow_1_0::NVIDIA_OF_PERF_LEVEL::NV_OF_PERF_LEVEL_FAST, false, false, false, 0);
+        NvidiaOpticalFlow_1_0::NVIDIA_OF_PERF_LEVEL::NV_OF_PERF_LEVEL_FAST, false, false, false, 0, inputStream, outputStream);
```

Comment:
API behavior is confusing.
Why we provide default values for parameters (see #2536), but still require them?

How this correlates with documentation?

> If input stream is not set, the execute function will use default stream which is NULL stream;

Reply:
We dont require them. 
Those are optional parameters. Default values are provided so that if user dont pass it then it will consider default value.
This sample app demonstrates comparison of different optical flow algorithms and we added streams to this so as to see the benefit of it in performance compared with other algorithms.